Zigbee2MqttAssistant/Services/MqttConnectionService.cs

```diff
@@ -135,57 +135,80 @@ CronExpression ParseCronExpression(string cronExpression)
-					}
-					catch (Exception ex)
+					while (!ct.IsCancellationRequested)
 					{
-						_logger.LogError(ex, "Error sending device request");
-					}
+						try
+						{
+							await SendDevicesRequest();
+						}
+						catch (Exception ex)
+						{
+							_logger.LogError(ex, "Error sending device request");
+						}
 
-					var now = DateTimeOffset.Now;
-					var next = cron?.GetNextOccurrence(now, TimeZoneInfo.Local) ?? now.AddMinutes(6);
-					var delay = next - now;
+						var now = DateTimeOffset.Now;
+						var next = cron?.GetNextOccurrence(now, TimeZoneInfo.Local) ?? now.AddMinutes(6);
+						var delay = next - now;
 
-					_logger.LogDebug($"PollingDevicesTask: Waiting until {next} (currently is {now}. (cron={cron})");
-					await Task.Delay(delay, ct);
+						_logger.LogDebug(
+							$"PollingDevicesTask: Waiting until {next} (currently is {now}. (cron={cron})");
+						await Task.Delay(delay, ct);
+					}
+				}
+				catch (TaskCanceledException)
+				{
+				}
+				catch (Exception ex)
+				{
+					_logger.LogError(ex, "PollingDevicesTask() shutting down because of an exception.");
 				}
 			}
 
 			async void PollingNetworkTask()
 			{
-				var cron = ParseCronExpression(_settings.CurrentSettings.NetworkScanSchedule);
+				try
+				{
+					var cron = ParseCronExpression(_settings.CurrentSettings.NetworkScanSchedule);
 
-				// A random offset is used to prevent many instances to send the same request
-				// exactly at the same time
-				var randomOffset = TimeSpan.FromSeconds(new Random().NextDouble() * 6);
+					// A random offset is used to prevent many instances to send the same request
+					// exactly at the same time
+					var randomOffset = TimeSpan.FromSeconds(new Random().NextDouble() * 6);
 
-				await Task.Delay(30000, ct);
+					await Task.Delay(30000, ct);
 
-				while (!ct.IsCancellationRequested)
-				{
-					try
-					{
-						await SendNetworkScanRequest();
-					}
-					catch (Exception ex)
+					while (!ct.IsCancellationRequested)
 					{
-						_logger.LogError(ex, "Error sending network scan request");
-					}
+						try
+						{
+							await SendNetworkScanRequest();
+						}
+						catch (Exception ex)
+						{
+							_logger.LogError(ex, "Error sending network scan request");
+						}
 
-					var now = DateTimeOffset.Now;
-					var next = (cron?.GetNextOccurrence(now, TimeZoneInfo.Local) ?? now.AddMinutes(20)) + randomOffset;
-					var delay = (next - now);
+						var now = DateTimeOffset.Now;
+						var next = (cron?.GetNextOccurrence(now, TimeZoneInfo.Local) ?? now.AddMinutes(20)) +
+						           randomOffset;
+						var delay = (next - now);
 
-					_logger.LogDebug($"PollingNetworkTask: Waiting until {next} (currently is {now}. (cron={cron})");
-					await Task.Delay(delay, ct);
+						_logger.LogDebug(
+							$"PollingNetworkTask: Waiting until {next} (currently is {now}. (cron={cron})");
+						await Task.Delay(delay, ct);
+					}
+				}
+				catch (TaskCanceledException)
+				{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either remove or fill this block of code.](https://app.codacy.com/gh/yllibed/Zigbee2MqttAssistant/pullRequest?prid=4988101)